src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -202,10 +202,11 @@ internal sealed class SourcePropertySymbol : PropertySymbol, IAttributeTargetSym
 
                 if (notRegularProperty)
                 {
-                    Binder.CheckFeatureAvailability(location,
-                                                    isReadOnly ? MessageID.IDS_FeatureReadonlyAutoImplementedProperties :
-                                                                 MessageID.IDS_FeatureAutoImplementedProperties,
-                                                    diagnostics);
+                    Binder.CheckFeatureAvailability(
+                        syntax,
+                        location,
```

Comment:
`location` is the location of the property name (or `this` for indexers) where as `syntax` is the entire property.